common/src/main/java/com/wynntils/services/map/PoiService.java

```diff
@@ -75,87 +74,21 @@ public class PoiService extends Service {
 
     public PoiService() {
         super(List.of());
-
-        loadData();
     }
 
     @Override
-    public void reloadData() {
-        loadData();
+    public void registerDownloads(DownloadRegistry registry) {
+        registry.registerDownload(UrlId.DATA_STATIC_PLACES).handleReader(this::handlePlaces);
+        registry.registerDownload(UrlId.DATA_STATIC_SERVICES).handleReader(this::handleServices);
+        registry.registerDownload(UrlId.DATA_STATIC_COMBAT_LOCATIONS).handleReader(this::handleCombat);
+        registry.registerDownload(UrlId.DATA_STATIC_CAVE_INFO).handleReader(this::handleCaves);
     }
 
     @Override
     public void onStorageLoad() {
         loadCustomPoiProviders();
     }
 
-    private void loadData() {
-        loadPlaces();
-        // Slightly hacky way to reduce risk of class loading race in development environment
-        // These are loaded serially after places instad
-        if (WynntilsMod.isDevelopmentEnvironment()) return;
-
-        loadCaves();
-        loadServices();
-        loadCombat();
-        loadCustomPoiProviders();
-    }
-
-    private void loadCaves() {
-        Download dl = Managers.Net.download(UrlId.DATA_STATIC_CAVE_INFO);
-        dl.handleReader(reader -> {
-            Type type = new TypeToken<List<CaveProfile>>() {}.getType();
-
-            List<CaveProfile> profiles = GSON.fromJson(reader, type);
-
-            cavePois.addAll(profiles.stream()
-                    .map(profile -> {
-                        CombatListProvider.registerFeature(profile.location, CombatKind.CAVES, profile.name);
-                        return new CombatPoi(
-                                PoiLocation.fromLocation(profile.location), profile.name, CombatKind.CAVES);
-                    })
-                    .collect(Collectors.toUnmodifiableSet()));
-        });
-    }
-
-    public void loadCustomPoiProviders() {
-        for (CustomPoiProvider poiProvider : customPoiProviders.get()) {
-            Managers.Net.download(poiProvider.getUrl(), poiProvider.getName()).handleJsonArray(elements -> {
-                List<CustomPoi> pois = new ArrayList<>();
-
-                for (JsonElement jsonElement : elements) {
-                    CustomPoi poi = GSON.fromJson(jsonElement, CustomPoi.class);
-                    pois.add(poi);
-                }
-
-                providedCustomPois.put(poiProvider, ImmutableList.copyOf(pois));
-            });
-        }
-    }
-
-    @SubscribeEvent
-    public void onDataLoaded(NetResultProcessedEvent.ForUrlId event) {
-        if (!WynntilsMod.isDevelopmentEnvironment()) return;
-        // Serialize loading of POIs when on dev env
```

Comment:
So this was just about triggering a race, which will be resolved by itself in the new system? Asking this as I see that you have added no dependency constraints for the DownloadManager. 

Or, to phrase it differently: are you sure that there are no dependencies between these downloads that you have missed to express?

Reply:
We really have two options here. The serialized loading for POIs originally happened because of a multi-thread class loading race between threads in the fork join pool. As far as I know it's been fixed in Arch-Transformer / EventBus (or Bus, as they call it now..). I can either add the dependency back, or what I would rather do, remove it, as it's really likely to be fixed, and even if it is not, it's a race that only happens in development environment, meaning users won't have issues with it.

Reply:
Right, it was some class loading stuff. Now I recall. Yeah, remove it, and hope it's been fixed. Otherwise we'll deal with it.